vmdb/app/models/host.rb

```diff
@@ -789,8 +789,9 @@ def refreshable_status
   def scannable_status
     s = self.refreshable_status
     if s[:show] == false && s[:enabled] == false
-      if self.authentication_valid?(:ipmi) || !self.ipmi_address.blank?
-        if self.authentication_invalid?(:ipmi)
+      # TODO: Simplify these conditions
+      if self.has_credentials?(:ipmi) || !self.ipmi_address.blank?
+        if self.missing_credentials?(:ipmi)
```

Comment:
| has impi creds | ipmi address set | then do |
| --- | --- | --- |
| true | true | "" |
| true | false | Provide an IPMI Address |
| false | true | Provide credentials for IPMI |
| false | false | falls through, so ???? |